pkg/artifactory/resource_artifactory_access_token.go

```diff
@@ -300,8 +300,10 @@ func unpackGroups(d *schema.ResourceData, client *resty.Client, tokenOptions *Ac
 		for i, group := range srcGroups.([]interface{}) {
 			groups[i] = group.(string)
 
-			if exist, err := checkGroupExists(client, groups[i]); !exist {
-				return err
+			if groups[i] != "*" {
+				if exist, err := checkGroupExists(client, groups[i]); !exist {
```

Comment:
This isn't quite right. On `err` return err. On `!exists` well, that is also an error but a totally different kind. And, I realize that the original logic did this, but I can see now that it is wrong.  I am going to look at the code now holistically

Reply:
yeah, even thought it isn't something you wrote, I now see this as a bug. So, if you could make a change and test, that'd be great.

```go
exist, err := checkGroupExists(client, groups[i])
if err {
   return err
}
if !exist {
   return fmt.Errorf("group %s doesn't exist",group)
}
```

If not, we can accept as is and I can ask a team member to make this change

Reply:
Ok, looking at this implementation of this `checkGroupExists` and it seems that 
`!exists == (err != nil)`
So, although I hate the optics here. It works. @alexhung We can probably merge